### ISSUE TITLE
SC1054 wrongly reported for func() {( … )}

### ISSUE THREAD

**Issue by blueyed** [koalaman/shellcheck#937]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC1054
- My shellcheck version (`shellcheck --version` or "online"): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

[SC1054](https://github.com/koalaman/shellcheck/wiki/SC1054) ("You need a space after the '{'.") gets reported for the following code on line 3, but that does not seem to be the case for both bash and dash:

```sh
#!/bin/dash

foo() {(
  echo 1
)}

foo
```

**Comment by koalaman** [comment]
Indeed, you're right. `(` is allowed in this context. This has now been fixed.

Of course, this could also have been written as

```
foo() (
  echo 1
)
```
